src/parser/github-comment-module.ts

```diff
@@ -203,6 +220,39 @@ export class GithubCommentModule extends BaseModule {
     return content.join("");
   }
 
+  async _isCollaborative(data: Readonly<IssueActivity>) {
+    if (!data.self?.closed_by || !data.self.assignee) return false;
+
+    const pullReview = data.linkedReviews[0];
```

Comment:
The problem here is that the reviews can still have "APPROVED" state even though afterward a "CHANGES_REQUESTED" is present. You should first split the reviews by assignee and remove the ones that are in the `review_requested` array from the pull-request itself. I had to implement a similar logic here: https://github.com/ubiquity-os-marketplace/command-start-stop/pull/91/files#diff-ad7752f33a7a7086e9ee9af78c95b792d0830df17b5f9be162e0b951bcd8e1d2R251
I do not know if there is a simpler approach using GraphQL.